### PR TITLE
Masked autoencoder pre-training for virtual staining models

### PR DESCRIPTION
**Pull request by ziw-liu**

### CODE REVIEW

**Review by edyoshikun — APPROVED** [review]
This branch has worked well for both 2.2D and 3D LUNeXT training/test so I am happy with it. 
- Combined and Concatenated data loader work
- Augmentations show no issue

**Comment by mattersoflight** [comment]
@ziw-liu Looks like this is now the de facto branch for both training recipes: end-to-end training and pre-training + fine-tuning. If it also provides a reasonably stable implementation of concatenated data loaders, please merge in `main`.

**Comment by ziw-liu** [comment]
> Looks like this is now the de facto branch for both training recipes: end-to-end training and pre-training + fine-tuning. Can you confirm if that is the case? Please merge in main if the merge won't break the current experiments.

This is the case. However note that I didn't do comprehensive backwards compatibility testing so it ***could*** break previously trained models.
